hpfeeds/broker/auth/mongo.py

```diff
@@ -0,0 +1,61 @@
+import asyncio
+import logging
+
+import motor.motor_asyncio
+from hpfeeds.broker.server import ServerException
+
+# inserting a user:
+# mongo
+# db.auth_key.insert({"identifier": "testin", "secret": "secretkey", "publish": ["chan1", "chan2"], subscribe: ["chan2"]})
+# Connection string example mongodb://username:password@mongohost:27017/dbname
+
+
+class Authenticator(object):
+
+    def __init__(self, connection_string):
+        self.logger = logging.getLogger('hpfeeds.broker.auth.mongo.Authenticator')
+        self.connection = False
+        try:
+            self.dbstring, self.dbname = connection_string.rsplit('/', 1)
+        except Exception as err:
+            ServerException("Unable to create connection string: {0}".format(err))
+
+    async def db_connect(self):
```

Comment:
It looks like this only gets called once, so you should be able to do this in `start`. This is called one inside a reactor right before the ports are opened.